bayes_opt/bayesian_optimization.py

```diff
@@ -120,6 +129,14 @@ def __init__(self, f, pbounds, random_state=None, verbose=2,
             random_state=self._random_state,
         )
 
+        if constraint is None:
+            # Data structure containing the function to be optimized, the bounds of
+            # its domain, and a record of the evaluations we have done so far
+            self._space = TargetSpace(f, pbounds, random_state)
+        else:
+            self._space = ConstrainedTargetSpace(f, constraint, pbounds, random_state)
+        self.constraint = constraint
```

Comment:
I notice a lot of these properties are prepended with a `_` and equipped with a separate method essentially acting as a `getter`, presumably to avoid users setting these properties. Let me know which properties you want me to handle this way.

Reply:
hey, I might consider putting self.constraint as self._constraint - which would be consistent with how _gp is currently handled.